### DISCUSSION TITLE
Aligning Map and PDB coordinate systems

### DISCUSSION THREAD

**Discussion by vratins** [project-gemmi/gemmi#341] [Q&A]
Hi,

I'm trying to label grid points from a CCP4 map based on closest atoms in the aligned PDB. My issue is that the coordinate systems of the PDB file and the Grid don't seem to align (after getting cartesian co-ordinates for each grid point after fractionalization). 

My current pipeline is: get fractional coordinates for each grid point -> use orthogonalization matrix to get cartesian coordinates for each grid point -> use nearest neighbour search between grid points and pdb coordinates to label a grid point based on atom type based on a specified radius.

The final step would only work if the map and PDB are aligned in their origin and coordinate systems (which they do not seem to be).

Is there a way to fix this or around this (or are there any flaws in my thought process) ?

Thanks!

**Comment by wojdyr** [comment]
If the map has the same unit cell as the PDB file, and it corresponds to this PDB file, I think it should work.
You can call `grid.get_position(i,j,k)` to get Cartesian coordinates corresponding to a grid point.

**Reply by vratins** [reply]
Thanks for the response! When I try to get the cartesian coordinates for every grid point (iterating over grid. nu,nv,nw) using `grid.get_position()` the values range from (0,0,0) to some (x,y,z) but the PDB file usually has its coordinates ranging from some negative (-x,-y,-z) to some positive (x,y,z) so the bounds seem misaligned. Is the way I'm obtaining all possible grid points incorrect?

**Reply by wojdyr** [reply]
Ah, that's what you meant. This is normal, you need to take symmetry into account when searching for nearest neighbors. If you use gemmi.NeighborSearch, it is handled automatically.

**Reply by vratins** [reply]
Just so I understand correctly, `gemmi.Position` is a cartesian coordinate, and `gemmi.Point` is a grid point, yes? Is there a way to use NeighborSearch to get the nearest atom to a `gemmi.Point`?

**Reply by wojdyr** [reply]
Yes, first you get Cartesian coordinates corresponding to the point
`pos = grid.point_to_position(point)`
and then
`ns.find_nearest_atom(pos)`
I'm assuming here that you have a system with crystallographic symmetry. (I don't remember if it also works for non-crystallographic models.)

**Reply by vratins** [reply]
Thank you! This works in finding the nearest atom to a grid point. How does NeighbourhoodSearch deal with unmodeled density? From my trials it seems it will find a nearest atom for every grid point?

**Reply by wojdyr** [reply]
NeighborSearch doesn't deal with density. It finds atoms in a model. For more details see:
https://gemmi.readthedocs.io/en/latest/analysis.html#neighbor-search
